middle_end/inlining_stats_types.ml

```diff
@@ -82,39 +104,80 @@ module Not_inlined = struct
     | With_subfunctions of Wsb.t * Wsb.t
 
 
-  let summary ppf = function
+  let summary ~rhyme ppf = function
     | Classic_mode ->
-      Format.pp_print_text ppf
-        "This function was prevented from inlining by `-Oclassic'."
-    | Function_obviously_too_large size ->
-      Format.pp_print_text ppf
-        "This function was not inlined because \
-         it was obviously too large";
-        Format.fprintf ppf "(%i)" size
+        begin match rhyme with
+        | `ion ->
+            Format.pp_print_string ppf
+              "This function was not inlined\n\
+               Indeed by the user it was declined\n\
+               When he specified the `-Oclassic' option\n"
+        | `ound ->
+            Format.pp_print_string ppf
+              "We did not look around\n\
+               To see if we could inline this function\n\
+               As the user specified the `-Oclassic' option\n"
+        end
+    | With_subfunctions _
+    | Without_subfunctions _
+    | Function_obviously_too_large _ ->
+        begin match rhyme with
+        | `ion ->
+            Format.pp_print_string ppf
+              "Compared to the code size evolution\n\
+               The expected benefit was to slim\n\
```

Comment:
To slim, to shorten, to shrink
The goal of this code, I think
To slim, beneficial
But too slim, superficial
A typo perhaps, rethink?